cmake/GenerateTestSchedule.cmake

```
@@ -0,0 +1,55 @@
+# generate_test_schedul(<output file> ...)
+#
+# A test schedule is generated for the files in TEST_FILES. The test schedule
+# groups the tests into groups of size GROUP_SIZE, with the exceptions of any
+# tests mentioned in the list SOLO, which will be in their own test group.
+#
+# TEST_FILES <file> ...
+#
+# Test files to generate a test schedule for.
+#
+# SOLO <test> ...
+#
+# Names of tests that should be run as solo tests. Note that these are test
+# names, not file names.
+#
+# GROUP_SIZE
+#
+# Size of each group in the test schedule.
+function(generate_test_schedule OUTPUT_FILE)
+  set(options)
+  set(oneValueArgs GROUP_SIZE)
+  set(multiValueArgs TEST_FILES SOLO)
+  cmake_parse_arguments(_generate "${options}" "${oneValueArgs}"
+                        "${multiValueArgs}" ${ARGN})
+
+  list(SORT _generate_TEST_FILES)
+  file(REMOVE ${OUTPUT_FILE})
+
+  # We put the solo tests in the test file first. Note that we do not generate
+  # groups for solo tests that are not in the list of test files.
+  foreach(_solo ${_generate_SOLO})
+    if("${_test}.sql" IN_LIST _generate_TEST_FILES)
```

```suggestion
    if("${_solo}.sql" IN_LIST _generate_TEST_FILES)
```
is this what you were trying to do?

---

Yup. Surprised that it passed the tests anyway.